docs/proposals/20200113-managed-kubernetes-provider.md

```diff
@@ -0,0 +1,93 @@
+
+**Potential benefits include:**
+- Faster cluster provisioning
+- Improved provisioning success rates
+- Automatic cluster operations for create/delete/upgrade/scaling if supported by cloud provider
+- Deeply integrated with cloud provider on storage, network, LB, security and HA
+- Improved user experience
+   - less YAML files to maintain
+
+### Goals
+
+-  Enhance existing CAPI providers for leveraging MKP to managing Kubernetes cluster.
+  - Add [EKS](https://aws.amazon.com/eks/) support to [AWS provider](https://github.com/kubernetes-sigs/cluster-api-provider-aws) 
+  - Add [IKS](https://www.ibm.com/cloud/container-service/) support to [IBM Cloud provider](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud)
+  - Add [AKS](https://azure.microsoft.com/en-in/services/kubernetes-service/) support to [Azure provider](https://github.com/kubernetes-sigs/cluster-api-provider-azure)
+  - Add [GKE](https://cloud.google.com/kubernetes-engine/) support to [GCP provider](https://github.com/kubernetes-sigs/cluster-api-provider-gcp)
+  - Add [DOKS](https://www.digitalocean.com/products/kubernetes/) support to [DigitalOcean provider](https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean)
+  - Add [ACK](https://www.alibabacloud.com/product/kubernetes) support to [Alibaba Cloud](https://github.com/oam-oss/cluster-api-provider-alicloud) 
+
+### Non-goals/Future Work
+
+- To integrate with the Kubernetes cluster autoscaler.
+
+## Proposal
+
+This proposal introduces MKP for the purpose of delegating the management of Kubernetes cluster to existing infrastructure [provider](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/providers.md#infrastructure), after that, the providers in `Goals` section will have 2 ways manage Kubernetes cluster according to different user case.
+
+It should have 3 approaches to extend existing [providers](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/providers.md#infrastructure) to have MKP functionality: 
+ 1.  Reuse `infrastructureRef`from `Cluster` but have new MKP implementation:  for example, [awscluster_controller.go](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/controllers/awscluster_controller.go) VS `ekscluster_controller.go`
+ 2.  Implement MKP based controller Control Plane to fill the gap from `Non-Goals/Future Work` section in [kubeadm-based-control-plane.md](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191017-kubeadm-based-control-plane.md), but cover both control plane and worker load plane.
+ 3.  (**I  preferred**)Create individual entity `managedK8SRef` to `Cluster` against `infrastructureRef`with new controller: for example, [awscluster_controller.go](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/controllers/awscluster_controller.go) VS `ekscluster_controller.go`
```

Comment:
Why do you prefer this option?

Reply:
Add new individual entity `managedK8SRef` plug-in to `Cluster` will make the code logic is clear and align with k8s declarative api design style, `infrastructureRef`  `ControlPlaneRef ` and `managedK8SRef` are different user case from it's name, my preferred option will has minim change from cluster api core and easy for existing provider to implementation and maintenance.

Reply:
I am not in favor of adding a field specifically for managed Kubernetes providers. They should be able to hook in either as ControlPlaneRefs or InfrastructureRefs. xref https://github.com/kubernetes-sigs/cluster-api/pull/2045#discussion_r366428652

Reply:
I'd also like to add my +1 that a generic name like ControlPlaneRef is highly preferable.